### PR TITLE
Fix IO error when requesting file with trailing slash

### PR DESCRIPTION
**Pull request by fmckeogh**
**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [ ] Have you created/updated the relevant documentation page(s)?

### CODE REVIEW

**Comment by Keats** [comment]
How does that happen in practice?

**Comment by fmckeogh** [comment]
@Keats I get panics if I create an empty Zola project, add a file called `logo.png` then try to access `/logo.png/`. Other websites (imgur, Google) appear to trim the trailing slash.

**Comment by fmckeogh** [comment]
@keats Screenshot 2022-10-17 at 19 55 12

**Comment by Keats** [comment]
Wouldn't `/image.png/` try to load `/image.png/index.html` in most webservers? `python3 -m http.server` has kinda the same behaviour as Zola here and 404

**Comment by fmckeogh** [comment]
@Keats  I only tested a handful of websites, honestly I'm happy with any behaviour (redirect, normalise all URLs, 404) other than crashing:)

**Comment by Keats** [comment]
Let's return a 404 then

**Comment by fmckeogh** [comment]
https://github.com/rust-lang/rust/issues/86442 tagging issue so that I can open another PR in the future when the feature is stabilised.

**Comment by fmckeogh** [comment]
@Keats Sorry to bother you, commit [b0b7a46](https://github.com/getzola/zola/pull/2006/commits/b0b7a46aa88a9cbcff9f1716efe498310ab64406) returns a 404 on trailing slash.

File: src/cmd/serve.rs [hunk 1]
```diff
@@ -182,11 +182,22 @@ fn method_not_allowed() -> Response<Body> {
...
         std::io::ErrorKind::NotFound => not_found(),
         std::io::ErrorKind::PermissionDenied => {
             Response::builder().status(StatusCode::FORBIDDEN).body(Body::empty()).unwrap()
         }
-        _ => panic!("{}", err),
+
+        _ => {
+            // "Not A Directory" IO error occurs when a request is made for an
+            // existing file with a trailing forward slash (e.g. `/logo.png/`)
+            if err.raw_os_error() == Some(20) {
```
Comment (Keats) [inline src/cmd/serve.rs:195]: Is that error number the same on all OSes?

Comment (fmckeogh) [reply src/cmd/serve.rs:195]: Tested on macOS and Linux, but yes, you're right, this is broken on Windows. Closing until https://github.com/rust-lang/rust/issues/86442 is stabilised.

**Pull request closed without merging**